### ISSUE TITLE
Local development setup documentation

### ISSUE THREAD

**Issue by whikloj** [danloveg/NCTR-Bagit-Record-Transfer#5]
Hi @danloveg,

Trying to follow the **Running Locally for Development** steps here (https://nctr-bagit-record-transfer.readthedocs.io/en/latest/running/local.html) and it seems some things are missing.

When bringing up the containers the `recordtransfer_rq_workers_dev` fails because I need to add an `ARCHIVIST_EMAIL=...` environment variable in the `.dockerenv`

Then I seem to need to create the sub-directories `media/bags` inside the `bagitobjecttransfer` based on the docker-compose.dev.yml mapping `./` to `/app`. I'm not fluent in docker so this might be an error on my side.

Once I've made those changes I still get the following issues with the Redis server and the worker application

```
> docker-compose -f docker-compose.dev.yml up  
Creating network "bagitobjecttransfer_default" with the default driver
Creating recordtransfer_mailhog_dev ... done
Creating redis_server_dev           ... done
Creating recordtransfer_rq_workers_dev ... done
Attaching to recordtransfer_mailhog_dev, redis_server_dev, recordtransfer_rq_workers_dev
recordtransfer_mailhog_dev | 2021/04/13 19:19:57 Using maildir message storage
recordtransfer_mailhog_dev | 2021/04/13 19:19:57 Maildir path is /home/mailhog/mail
redis_server_dev | 
redis_server_dev | *** FATAL CONFIG FILE ERROR ***
redis_server_dev | Reading the configuration file, at line 108
redis_server_dev | >>> 'logfile /var/log/redis/redis-server.log'
redis_server_dev | Can't open the log file: Is a directory
recordtransfer_mailhog_dev | [HTTP] Binding to address: 0.0.0.0:8025
recordtransfer_mailhog_dev | 2021/04/13 19:19:57 Serving under http://0.0.0.0:8025/
recordtransfer_mailhog_dev | 2021/04/13 19:19:57 [SMTP] Binding to address: 0.0.0.0:1025
recordtransfer_mailhog_dev | Creating API v1 with WebPath: 
recordtransfer_mailhog_dev | Creating API v2 with WebPath: 
redis_server_dev exited with code 1
recordtransfer_rq_workers_dev | Error 111 connecting to 0.0.0.0:6379. Connection refused.
recordtransfer_rq_workers_dev exited with code 1
```

I'll note I'm running this on OSX which has in past caused me some docker headaches, but if you might have a suggestion I'd appreciate it.

**Comment by danloveg** [comment]
Hi,

This application is _very_ much still a work in progress, but I'm sorry you're running into so many issues trying to run it. The `redis_server_dev` container seems to think that `/var/log/redis/redis-server.log` is a directory. Since this is declared in `docker-compose.dev.yml` as linking to `logs/redis-server.log` in your current directory, it looks like Docker created a folder for you called `redis-server.log` in the logs folder instead of a file.

Try deleting the folder `logs/redis-server.log` and creating an empty file there with the same name.

**Comment by whikloj** [comment]
I can do that, should I do the same for the other two?

```
[/Users/whikloj/www/NCTR/NCTR-Bagit-Record-Transfer/bagitobjecttransfer/logs] 
> ls -l  
total 0
drwxr-xr-x  2 whikloj  staff  64 13 Apr 13:26 recordtransfer.log
drwxr-xr-x  2 whikloj  staff  64 13 Apr 14:19 redis-server.log
drwxr-xr-x  2 whikloj  staff  64 13 Apr 13:26 rqworker.log
```

**Comment by danloveg** [comment]
Yes, those should all be files.

**Comment by whikloj** [comment]
Looks like providing file paths in a `volume` option can cause problems if they don't exist to start.

https://github.com/docker/compose/issues/2781#issuecomment-501654001

I did this instead.
```
> git diff docker-compose.dev.yml 
diff --git a/bagitobjecttransfer/docker-compose.dev.yml b/bagitobjecttransfer/docker-compose.dev.yml
index bce655b..3549076 100644
--- a/bagitobjecttransfer/docker-compose.dev.yml
+++ b/bagitobjecttransfer/docker-compose.dev.yml
@@ -22,8 +22,8 @@ services:
     volumes:
       - ./:/app
       - /tmp:/tmp
-      - ./logs/rqworker.log:/var/log/django-rq/rqworker.log
-      - ./logs/recordtransfer.log:/var/log/django/recordtransfer.log
+      - ./logs/django:/var/log/django-rq
+      - ./logs/django:/var/log/django
     env_file:
       - ./.dockerenv
     environment:
@@ -39,7 +39,7 @@ services:
       - 6379:6379
     volumes:
       - ./redis/redis.conf:/usr/local/etc/redis/redis.conf
-      - ./logs/redis-server.log:/var/log/redis/redis-server.log
+      - ./logs/redis:/var/log/redis
     environment:
       REDIS_REPLICATION_MODE: master
```

I still have a stray `./logs/redis-server.log` but the application seems to start.

**Comment by danloveg** [comment]
Great, I'll address this issue in an upcoming commit.

**Comment by danloveg** [comment]
Closed by #10, thanks @whikloj
